sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/agg/AggregationCombineFnAdapter.java

```diff
@@ -17,77 +17,146 @@
+    MultiInputCombiner(CombineFn<Row, Object, Object> combineFn) {
+      super(combineFn);
+    }
+
+    @Override
+    Row getInput(Row input) {
+      for (Object o : input.getValues()) {
+        if (o == null) {
+          return null;
+        }
+      }
+      return input;
+    }
   }
 
-  /**
-   * Creates an instance of {@link AggregationCombineFnAdapter}.
-   *
-   * @param callWithAlias Calcite's output, represents a function call paired with its field alias
-   */
-  public static AggregationCombineFnAdapter of(
-      Pair<AggregateCall, String> callWithAlias, Schema inputSchema) {
-    AggregateCall call = callWithAlias.getKey();
-    Schema.Field field = CalciteUtils.toField(callWithAlias.getValue(), call.getType());
-    String functionName = call.getAggregation().getName();
-
-    return new AggregationCombineFnAdapter(
-        field,
-        createCombineFn(call, field, functionName),
-        AggregationArgsAdapter.of(call.getArgList(), inputSchema));
+  private static class SingleInputCombiner extends WrappedCombinerBase<Object> {
+    SingleInputCombiner(CombineFn<Object, Object, Object> combineFn) {
+      super(combineFn);
+    }
+
+    @Override
+    Object getInput(Object input) {
+      return input;
+    }
+  }
+
+  private static class ConstantEmpty extends CombineFn<Row, Row, Row> {
+    private static final Schema EMPTY_SCHEMA = Schema.builder().build();
+    private static final Row EMPTY_ROW = Row.withSchema(EMPTY_SCHEMA).build();
+
+    public static final ConstantEmpty INSTANCE = new ConstantEmpty();
+
+    @Override
+    public Row createAccumulator() {
+      return EMPTY_ROW;
+    }
+
+    @Override
+    public Row addInput(Row accumulator, Row input) {
+      return EMPTY_ROW;
+    }
+
+    @Override
+    public Row mergeAccumulators(Iterable<Row> accumulators) {
+      return EMPTY_ROW;
+    }
+
+    @Override
+    public Row extractOutput(Row accumulator) {
+      return EMPTY_ROW;
+    }
+
+    @Override
+    public Coder<Row> getAccumulatorCoder(CoderRegistry registry, Coder<Row> inputCoder)
+        throws CannotProvideCoderException {
+      return SchemaCoder.of(EMPTY_SCHEMA);
+    }
+
+    @Override
+    public Coder<Row> getDefaultOutputCoder(CoderRegistry registry, Coder<Row> inputCoder) {
+      return SchemaCoder.of(EMPTY_SCHEMA);
+    }
   }
 
   /** Creates either a UDAF or a built-in {@link CombineFn}. */
-  private static CombineFn<?, ?, ?> createCombineFn(
+  public static CombineFn<?, ?, ?> createCombineFn(
       AggregateCall call, Schema.Field field, String functionName) {
+    CombineFn combineFn;
     if (call.getAggregation() instanceof SqlUserDefinedAggFunction) {
-      return getUdafCombineFn(call);
+      combineFn = getUdafCombineFn(call);
+    } else {
+      combineFn = BeamBuiltinAggregations.create(functionName, field.getType().getTypeName());
+    }
+    if (call.getArgList().isEmpty()) {
+      return new SingleInputCombiner(combineFn);
```

Comment:
`ConstantEmpty` here?

Reply:
No, that is not the same semantics (different types)